tests/test_base.py

```diff
@@ -58,9 +71,59 @@ def test_acknowledge_messages_call(mocked_subscriber_client, mocked_retry):
     )
 
 
+@mock.patch("pydrinker_gcp.base._get_subscriber")
 @mock.patch("pydrinker_gcp.base.pubsub_v1.SubscriberClient")
-def test_close_call(mocked_subscriber_client):
+def test_close_call(mocked_subscriber_client, mocked_get_subscriber):
+    mocked_get_subscriber.return_value = mocked_subscriber_client()
+
     base_sub = BaseSubscriber(project_id="xablau-xebleu-123456", subscription_id="sample-sub")
     base_sub.close()
 
     mocked_subscriber_client.return_value.close.assert_called_once_with()
+
+
+@mock.patch.dict(os.environ, {"GOOGLE_APPLICATION_CREDENTIALS": "credential.json"})
+@mock.patch("pydrinker_gcp.base.pubsub_v1.SubscriberClient")
+@mock.patch("pydrinker_gcp.base.os.path.isfile")
+def test_get_subscriber_with_credential_file(mocked_isfile, mocked_subscriber_client):
+    mocked_isfile.return_value = True
+    subscriber_client = _get_subscriber()
+
+    mocked_subscriber_client.assert_called_once_with()
+    assert subscriber_client == mocked_subscriber_client()
+
+
+@mock.patch("pydrinker_gcp.base.jwt.Credentials.from_service_account_info")
+@mock.patch("pydrinker_gcp.base.pubsub_v1.SubscriberClient")
+def test_get_subscriber_with_service_account_value(
+    mocked_subscriber_client, mocked_from_service_account_info
+):
+    google_service_account = (
+        '{"type": "service_account", "project_id": "fake-project-123456", "private_key_id": "123"}'
+    )
+    with mock.patch.dict(os.environ, {"GOOGLE_SERVICE_ACCOUNT": google_service_account}):
+        subscriber_client = _get_subscriber()
+
+        mocked_from_service_account_info.assert_called_once_with(
+            json.loads(google_service_account), audience=SUB_AUDIENCE
+        )
+        mocked_subscriber_client.assert_called_once_with(credentials=mocked_from_service_account_info())
+        assert subscriber_client == mocked_subscriber_client()
+
+
+@mock.patch("pydrinker_gcp.base.jwt.Credentials.from_service_account_info")
+@mock.patch("pydrinker_gcp.base.pubsub_v1.SubscriberClient")
+def test_get_subscriber_without_any_environment_variable(
```

Comment:
this test does the same as `test_get_subscriber_with_service_account_value`. I believe in this one you want to check whether `ProviderError` is raised

Reply:
Yes, this is wrong, i'll be fix that.